tower_cli/resources/node.py

```diff
@@ -0,0 +1,159 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, unicode_literals
+
+from tower_cli import models, resources
+from tower_cli.utils import types
+from tower_cli.utils.resource_decorators import unified_job_template_options
+from tower_cli.utils import exceptions, debug
+from tower_cli.api import client
+
+import click
+
+
+NODE_STANDARD_FIELDS = [
+    'unified_job_template', 'inventory', 'credential', 'job_type',
+    'job_tags', 'skip_tags', 'limit'
+]
+JOB_TYPES = {
+    'job': 'job_template',
+    'project_update': 'project',
+    'inventory_update': 'inventory_source'
+}
+
+
+class Resource(models.Resource):
+    cli_help = 'Manage nodes inside of a workflow job template.'
+    endpoint = '/workflow_job_template_nodes/'
+    identity = ('id',)
+
+    workflow_job_template = models.Field(
+        key='-W', type=types.Related('workflow'), required=False)
+    unified_job_template = models.Field(required=False)
+    inventory = models.Field(
+        type=types.Related('inventory'), required=False, display=False)
+    credential = models.Field(
+        type=types.Related('credential'), required=False, display=False)
+    job_type = models.Field(required=False, display=False)
+    job_tags = models.Field(required=False, display=False)
+    skip_tags = models.Field(required=False, display=False)
+    limit = models.Field(required=False, display=False)
+
+    def __getattribute__(self, attr):
+        method = super(Resource, self).__getattribute__(attr)
+        if attr in ['create', 'modify', 'list']:
+            return unified_job_template_options(method)
+        return method
+
+    @staticmethod
+    def _forward_rel_name(rel):
+        return '{0}_nodes'.format(rel)
+
+    @staticmethod
+    def _reverse_rel_name(rel):
+        return 'workflowjobtemplatenodes_{0}'.format(rel)
+
+    def _parent_filter(self, parent, relationship, **kwargs):
+        """
+        Returns filtering parameters to limit a search to the children
+        of a particular node by a particular relationship.
+        """
+        if parent is None or relationship is None:
+            return {}
+        parent_filter_kwargs = {}
+        query_params = ((self._reverse_rel_name(relationship), parent),)
+        parent_filter_kwargs['query'] = query_params
+        if kwargs.get('workflow_job_template', None) is None:
+            parent_data = self.read(pk=parent)['results'][0]
+            parent_filter_kwargs['workflow_job_template'] = parent_data[
+                'workflow_job_template']
+        return parent_filter_kwargs
+
+    @unified_job_template_options
```

Comment:
Good find on the help text.

But the problem isn't here. It was overriding the __getattribute__ that caused this problem... I hoped that would make life simple, but I have dived into more complexity than I bargained for.